### PR TITLE
oauth docs

### PR DESCRIPTION
**Pull request by mattlub**
- [x] moves api docs to folder
- [x] fixes category links in api docs #133
- [x] creates oauth flow docs #103 

heavily based on https://developer.github.com/apps/building-integrations/setting-up-and-registering-oauth-apps/about-authorization-options-for-oauth-apps/

### CODE REVIEW

**Review by m4v15 — APPROVED** [review]

File: docs/oauth.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+GET https://nazareth-open-tourism-platform.herokuapp.com/oauth/authorize
+```
+
+**Query parameters**
+
+Name | Type | Description
+--- | --- | ---
+client_id | string | **Required.** The client ID you received from Nazareth Open Tourism Platform for your app.
+redirect_uri | string | **Required.** The URL in your application where users will be sent after authorization.
+state | string | **Required.** An unguessable random string. It is used to protect against cross-site request forgery attacks.
```
Comment (des-des) [inline docs/oauth.md]: Could we get an example of the url with the query params

File: docs/oauth.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+### 2. Users are redirected back to your site by Nazareth Open Tourism Platform
+
+If the user accepts your request, Nazareth Open Tourism Platform redirects back to your site with a temporary authorizaion code in a `code` parameter as well as the state you provided in the previous step in a `state` parameter. If the states don't match, the request was created by a third party and the process should be aborted.
```
Comment (des-des) [inline docs/oauth.md]: > temporary authorizaion code in a `code` parameter

better: temporary authorizaion code in a `code` query parameter

could we have an example a url that the user might be sent to here

Comment (des-des) [reply docs/oauth.md]: > third party

maybe say malicious

File: docs/oauth.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+Exchange the `code` for an access token:
```
Comment (des-des) [inline docs/oauth.md]: Can you be more descriptive.

Ie once the user is redirected to your server, your server should make a post request over *https* to the url: .

Then add example.

**Review by des-des — CHANGES_REQUESTED** [review]
Okay cool.

Right now you need to talk a bit about actually creating secret, id etc.

Say that they need to get in touch with you to generate this stuff

**Comment by mattlub** [comment]
@des-des added how to get client id etc.
made other changes to improve clarity

**Comment by m4v15** [comment]
Simple example of the oauth flow from the client side here https://github.com/m4v15/OTP_Oauth
 - maybe we can have the app.js file in the docs?

**Comment by mattlub** [comment]
@m4v15 I would say link to it somewhere in our documentation.

**Review by des-des — APPROVED** [review]

**Comment by des-des** [comment]
Happy for you too merge this more stuff can be added in another pull if there is more you want to add

**Comment by m4v15** [comment]
@mattlub I fixed the merge conflicts (there weren't any locally for me, I don't really understand what happened). I then changed the GET and POST clients docs to be in the auth docs and secured them so that only SUPER's can post and view clients (for now)

**Comment by m4v15** [comment]
@mattlub cos I've made some changes I've left it for you to merge

**Comment by mattlub** [comment]
@m4v15 should that SUPER user stuff really be in the docs? I was under the impressions super users would literally only be us/the product owner

**Comment by m4v15** [comment]
@mattlub good point, moved

**Pull request merged**